tests/ff_module-task-event-editor.test.js

```diff
@@ -234,5 +247,71 @@ describe('TaskEventEditor', function() {
         });
     });
 
+    it('should fire correct handlers', function() {
+        events.forEach(function(event) {
+
+            var store = createStore(combineReducers({
+                event: modelReducer('event', event),
+                eventForm: formReducer('event', event)
+            }));
+
+            var spies = {
+                onSend: sinon.spy(),
+                onChange: sinon.spy(),
+                onClose: sinon.spy(),
+                onNext: sinon.spy()
+            }
+
+            function mapStateToProps(state) {
+                return {
+                    event: state.event,
+                    eventForm: state.eventForm,
+                    validation: validation,
+                    models: models,
+                    onSend: spies.onSend,
+                    onChange: spies.onChange,
+                    onClose: spies.onClose,
+                    onNext: spies.onNext
+                };
+            }
+
+            var ConnectedEditForm = connect(mapStateToProps)(TaskEventEditor);
+            var ProvidedForm = <Provider store={store}>
+                    <ConnectedEditForm />
+                </Provider>;
+
+            var component = TestUtils.renderIntoDocument(ProvidedForm);
+            var typeName = event.description.type;
+
+            if (event.state.error) typeName = eventStates.error;
+
+            var topCloseBtn = TestUtils.findRenderedDOMComponentWithClass(component, 'ff_container-dialog__close-top');
+            expect(topCloseBtn).to.exist;
+            TestUtils.Simulate.click(topCloseBtn);
+            expect(spies.onClose.calledOnce).to.equal(true);
+
+            var classes = buttonClasses[typeName];
+            classes.forEach(function(classDef){
+                var index = classDef[0],
+                    className = classDef[1],
+                    spyName = classDef[2];
+
+                var buttonList = TestUtils.scryRenderedDOMComponentsWithClass(component, className);
+                var button = buttonList[index];
+                expect(button).to.exist;
+
+                TestUtils.Simulate.click(button);
+                if (spyName === 'onClose') {
+                    expect(spies[spyName].calledTwice).to.equal(true);
+                } else  {
+                    expect(spies[spyName].calledOnce).to.equal(true);
+                }
+
+            })
+
+
+        });
```

Comment:
It might be worth extracting some parts of this test out into functions in order to keep the length down. I think that'll make it easier to read.